src/main/groovy/com/github/alisiikh/scalastyle/ScalastyleExtension.groovy

```diff
@@ -114,6 +123,11 @@ class ScalastyleExtension extends CommonScalastyleConfig {
         })
     }
 
+    @Inject
+    protected JavaForkOptionsFactory getJavaForkOptionsFactory() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
This getter looks like a hack, can't we use a private property to inject it?

Reply:
That's a good point. I've tried it out. Unfortunately, it seems the current setup is the only one in which it works.

Here are the things I've tried:
- `@Inject private final JavaForkOptionsFactory forkOptionsFactory`
- `@Inject private JavaForkOptionsFactory forkOptionsFactory`
- `@Inject protected JavaForkOptionsFactory forkOptionsFactory`
- `@Inject private JavaForkOptionsFactory getForkOptionsFactory() { ... }`

So it seems it neither can be a field nor private getter :(

I've borrowed the way in which it is implemented from the [official gradle `Test` task implementation](https://github.com/gradle/gradle/blob/v6.0.1/subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java#L179).

So for now, I've renamed the getter (`getJavaForkOptionsFactory` -> `getForkOptionsFactory`) and added a comment on where the inspiration for implementation comes from.